### PR TITLE
[MLU] Add cpu_allocator notifier protect and utilize hostmalloc for a…

### PR DESCRIPTION
**Pull request by PeiyuLau**
Add cpu_allocator notifier protect and utilize hostmalloc for asynH2D acceleration

### CODE REVIEW

File: backends/mlu/kernels/funcs/mlu_funcs.h [hunk 1]
```diff
@@ -63,14 +63,16 @@ inline void TensorCopy(const Context& dev_ctx,
     if (blocking) {
       MemCpyH2D(nullptr, dst_ptr, src_ptr, size);
     } else {
-      AsyncMemCpyH2D(nullptr, stream, dst_ptr, src_ptr, size);
+      C_Device_st device{dst_place_.GetDeviceId()};
```
Comment (ShawnNew) [inline backends/mlu/kernels/funcs/mlu_funcs.h]: 1. move this variable before if branch for code simplification
2. add all devices

**Review by ShawnNew — CHANGES_REQUESTED** [review]

File: backends/mlu/kernels/funcs/mlu_funcs.h (hunk 1 above)
Comment (PeiyuLau) [reply backends/mlu/kernels/funcs/mlu_funcs.h]: done

**Review by ShawnNew — APPROVED** [review]
LGTM

**Review by ronny1996 — APPROVED** [review]
LGTM

**Pull request merged**